apps/remote_control/test/lexical/remote_control/code_intelligence/definition_test.exs

```diff
@@ -0,0 +1,370 @@
+      uses_file = open_uses_file(project, uses_content)
+      position = cursor_to_position("hello|_func_in_using", uses_file)
+
+      assert {:ok, {source_file, range}} = Definition.definition(uses_file, position)
+
+      # credo:disable-for-next-line Credo.Check.Design.TagTODO
+      # TODO: this is wrong, it should go to the definition in the quote block
+      # but it goes to the `use` keyword in the caller module
+      # it can be fixed when we have a tracer, the tracer event kind will be `local_function`
+      assert annotate(source_file, range) == ~q[
+        defmodule MyModule do
+        ...
+
+          use MyDefinition
+        # ^^^
+      ]
+    end
+  end
+
+  describe "definition/2 when making local call" do
+    setup [:with_navigation_project, :with_referenced_file]
+
+    test "find the function definition", ctx do
+      {:ok, uses_file} = SourceFile.Store.open_temporary(ctx.uri)
+      position = cursor_to_position(~s[greet|("world")], uses_file)
+
+      {:ok, {source_file, range}} = Definition.definition(uses_file, position)
+
+      assert annotate(source_file, range) == ~q[
+        defmodule MyDefinition do
+        ...
+
+          def greet(name) do
+        #     ^^^^^
+      ]
+    end
+
+    test "find the attribute", ctx do
+      {:ok, uses_file} = SourceFile.Store.open_temporary(ctx.uri)
+      position = cursor_to_position("      @|b", uses_file)
+
+      {:ok, {source_file, range}} = Definition.definition(uses_file, position)
+
+      assert annotate(source_file, range) == ~q[
+        defmodule MyDefinition do
+        ...
+
+          @b 2
+        # ^^
+      ]
+    end
+
+    test "find the variable", ctx do
+      {:ok, uses_file} = SourceFile.Store.open_temporary(ctx.uri)
+      position = cursor_to_position("      a|", uses_file)
+
+      {:ok, {source_file, range}} = Definition.definition(uses_file, position)
+
+      assert annotate(source_file, range) == ~q[
+        defmodule MyDefinition do
+        ...
+
+            a = 1
+        #   ^
+      ]
+    end
+
+    test "can't find the definition when call a Elixir std module function", ctx do
+      {:ok, uses_file} = SourceFile.Store.open_temporary(ctx.uri)
+      position = cursor_to_position("String.to_intege|", uses_file)
+
+      # credo:disable-for-next-line Credo.Check.Design.TagTODO
+      # TODO: this should be fixed when we have a call tracer
+      {:ok, nil} = Definition.definition(uses_file, position)
+    end
+
+    test "find the definition when call a erlang module", ctx do
+      {:ok, uses_file} = SourceFile.Store.open_temporary(ctx.uri)
+      position = cursor_to_position("binary_to_|", uses_file)
+
+      {:ok, {source_file, range}} = Definition.definition(uses_file, position)
+
+      assert annotate(source_file, range) == ~q[
+        %%
+        ...
+
+        binary_to_atom(Binary) ->
+        ^^^^^^^^^^^^^^
+      ]
+    end
+  end
+
+  defp annotate(source_file, range) do
```

Comment:
this approach makes several assumptions that will bite us in the future, one of which is that the first line of a file is a module definition. Using CursorSupport will make doing this not necessary.

Reply:
Removed